### PR TITLE
rtorrent module

### PR DESCRIPTION
**Pull request by wrayan**
First try of a rtorrent module based on Chris Lucas' rtorrent-python api and the utorrent module code. 

Source doesn't look pretty and the template definately needs some more love but basic features are supported:

- connect to rtorrent (scgi url tested)
- list all torrents with the following information
  - torrent name
  - download progress bar, percentage, time remaining
  - status (seeding/busy/done/inactive)
  - seed ratio

### CODE REVIEW

**Comment by JedMeister** [comment]
@gugahoi is there a reason why this hasn't been merged? 

I'm planning on giving it a test anyway but I would be much rather be running your main branch than a fork. 

Or is there a fundamental reason why it can't or won't?

**Comment by JedMeister** [comment]
@wrayan I am not having luck with this. I suspect that it may be something that I have done wrong... In the module it says `connection error` and in the Maraschino logs it says `:: DEBUG    :: RTORRENTDL :: EXCEPTION - mismatched tag: line 6, column 2`

I would investigate further but I have no idea where to start looking (which file has a mismatched tag on line 6, col 2?)... Any ideas?

FWIW I am using nginx to provide xmlrpc via https on port 12322 (i.e. `https://127.0.0.1:12322/RPC2`). I assume that it is configured right as ruTorrent can connect ok... I also tried configuring rtorrent xmlrpc to use port 5000 (it was using a unix socket).

**Comment by JedMeister** [comment]
@wrayan after digging a little more I think my issue is caused by access via https while using a self signed cert...

**Comment by gugahoi** [comment]
I am sorry, completely forgot about it @JedMeister. Is it working for you?

**Comment by JedMeister** [comment]
NP @gugahoi. 

I couldn't get it to work directly via rTorrent sgci (I was using a unix socket which this code doesn't appear to support; but also tested after reconfiguring it to use localhost port 5000). 

Nor could I get it to work via the existing https port that I was already using for ruTorrent (web frontend for rTorrent). 

But I configured a new Nginx site to provide /RPC2 via localhost http:
````
server {
        listen 127.0.0.1:5000 default_server;
        listen [::1]:5000 default_server;

        server_name localhost;

	location /RPC2 {
		scgi_pass unix:/var/run/rtorrent/rpc.socket;
		include     scgi_params;
		scgi_param SCRIPT_NAME /RPC2;
	}
}
````
Then it all worked fine. I imagine that if I had a proper (CA signed) https cert then https would work too. AFAIK by default Python requires CA https certs (fails on self signed) but I could be wrong and perhaps there is some other issue.

Bottom line, yes it works. It could possibly do with a polish, but I don't know python very well nor do I know the Maraschino code so I won't be able to do that anytime soon. IMO it provides value as is.

**Comment by gugahoi** [comment]
Exactly what I was hoping to hear :smile:

**Comment by JedMeister** [comment]
Awesome. Thanks! :smile:

**Comment by wrayan** [comment]
@JedMeister thanks for reporting this. You're right, ssl doesn't seem to work at all. I'll look into it.
